system/churchcore/churchcore_db.php

```diff
@@ -1248,7 +1254,21 @@ function churchcore_delFile($id) {
   db_query("DELETE FROM {cc_file}
             WHERE id=:id",
             array (":id" => $id), false);
-  if (!unlink("$files_dir/files/$res->domain_type/$res->domain_id/$res->filename")) throw new CTFail("Datei konnte auf dem Server nicht entfernt werden.");
+
+  if (preg_match('/^[0-9a-f]{64}$/i', $res->filename)) {
+    $c = db_query("SELECT COUNT(*) AS cnt FROM {cc_file} WHERE filename = ?",
+        array ($res->filename), false)->fetch();
+    if((int)$c->cnt === 0) {
+      if (!unlink("$files_dir/blobs/$res->filename")) {
+        throw new CTFail("Datei konnte auf dem Server nicht entfernt werden.");
+      }
+    }
+  } else {
+    if (!unlink("$files_dir/files/$res->domain_type/$res->domain_id/$res->filename")) {
+      throw new CTFail("Datei konnte auf dem Server nicht entfernt werden.");
```

Comment:
Right, but I don't see a better option here than repeating this line, because the (pre)conditions of line 1263 and 1268 differ a lot.
Only possible solution that comes to my mind would be to throw a dummy-Exception, wrap everything in a try-catch-construct and throw the CTFail there... not really much nicer. ^^